File: A2/Sem4/MPP/Teme/TemaCS/TemaCS/domain/Entity.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TemaCS.domain
{
    internal class Entity<T> : IEquatable<Entity<T>>, IEqualityComparer<Entity<T>>
    {
        public T ID { get; set; }

        public Entity(T id)
        { 
            this.ID = id; 
        }

        public bool Equals(Entity<T>? other) => other != null && other.ID != null && other.ID.Equals(ID);

        public bool Equals(Entity<T>? x, Entity<T>? y) => 
            x == null && y == null || x != null && y != null && Equals(x.ID, y.ID);

        public int GetHashCode([DisallowNull] Entity<T> obj) => obj.ID.GetHashCode();

        public override string ToString() => $"ID: {ID}";
    }
}

File: A2/Sem3/MAP/Seminar/S6/src/domain/Entity.java
package domain;

public class Entity <ID>{
    protected ID id;

    public Entity() {
    }

    public Entity(ID id) {
        this.id = id;
    }

    public ID getId() {
        return id;
    }

    public void setId(ID id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return "Entity [id=" + id + "]";
    }
}
